### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by boo-shehab**
# Hi Reviewer, 👋

in this PR I used Air pollution API for my project, and this is a [video](https://www.loom.com/share/736c774cb5574c2cb2babbcb4fe375f7) to explain the project 
I did all the requirements
### Project requirements

**API**
First you need to choose an API to base the development of the webapp on. The API should allow you to get numeric values based on a parameter. We recommend that you choose an API that is open or needs an API key. If you choose an API that require authentication, **you should implement it on your own**. 

  Some example APIs are:
  - [Financial modeling prep API](https://financialmodelingprep.com/developer/docs/): stock data.
  - [Air pollution API](https://openweathermap.org/api/air-pollution): air quality data.

  You can find many APIs in [this GitHub repo](https://github.com/public-apis/public-apis) or in [ProgrammableWeb](https://www.programmableweb.com/category/all/apis). Some of the APIs requires a token for authentication and some others are just open.

**Design**
- You should follow these [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), including:
  - Colors (select one main color for your website).
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

Original design idea by [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

**Interactions**
- Home page
  - When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
  - Along with the category name, you will display some numeric values per category that come from the API.
  - When the user clicks (or taps) on a category item, the application navigates to the details page.

- Details page
  - In the details page, the webapp retrieves data from the API to show detailed data in the category.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

**Testing requirements**
- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
  - You may need to mock the access to the API, so that your tests don't send actual requests.
  - You may need to mock the connection to the Redux Store.

**Technical requirements**

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- You should filter the data that you retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project should be deployed and accessible online (add a link to the online version of your app to your README file).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,157 @@
...
+```sh
+  git clone https://github.com/boo-shehab/Metrics-webapp.git
+```
+
+### Install
+
+Install this project with:
+
+```sh
+  cd Math Magicians
```
Comment (Whoistolu) [inline README.md]: - [ ] [REQUIRED] Let's change directory (cd) to `Metrics Webapp` and not `Math Magicians` so as not to mislead the users of this application.

File: README.md [hunk 2]
```diff
@@ -1 +1,157 @@
...
+
+## ⭐️ Show your support <a name="support"></a>
+
+If you like this project please leave a star🤩
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
```
Comment (Whoistolu) [inline README.md]: - [ ] [REQUIRED] One of the project requirements states that the "Acknowledgements" section should give appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/). Let's fix this.

File: README.md [hunk 3]
```diff
@@ -1 +1,157 @@
...
+    <li><a href="#-acknowledgments-">🙏 Acknowledgments </a></li>
+    <li><a href="#-license-">📝 License</a></li>
+  </ul>
+</details>
+
+<br>
+
+<!-- PROJECT DESCRIPTION -->
+
+# 📖 Metrics webapp <a name="about-project"></a>
```
Comment (Whoistolu) [inline README.md:24]: - [ ] [REQUIRED] It is a requirement that you record a video presenting the features of the project. The maximum length of the video is 5 minutes. The video should contain a clear and objective description of the project in professional English. The video contains a demo of the project, highlighting all the project features. The video contains a highlight about a piece of the code that you are proud of making.

File: README.md [hunk 4]
```diff
@@ -1 +1,157 @@
...
+- LinkedIn: [LinkedIn](https://www.linkedin.com/in/ahmed-al-ali-77b6a0246/)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- FUTURE FEATURES -->
+
+## 🔭 Future Features <a name="future-features"></a>
+
+- there is no future right now
+
```
Comment (Whoistolu) [inline README.md]: - Kindly include 1-3 potential features to be added to this project. This will give a better knowledge of the application to the user.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @boo-shehab 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Ahmed, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- linters are passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**